src/libs/GooglePlacesUtils.js

```diff
@@ -10,18 +10,24 @@ import _ from 'underscore';
  * }
  *
  * @param {Array} addressComponents
- * @param {String} type
- * @param {String} key
- * @returns {String|undefined}
+ * @param {Object} fieldsToExtract – has shape: {addressType: 'keyToUse'}
+ * @returns {Object}
  */
-function getAddressComponent(addressComponents, type, key) {
-    return _.chain(addressComponents)
-        .find(component => _.contains(component.types, type))
-        .get(key)
-        .value();
+function getAddressComponents(addressComponents, fieldsToExtract) {
+    const result = _.mapObject(fieldsToExtract, () => '');
+    _.each(addressComponents, (addressComponent) => {
+        _.each(addressComponent.types, (addressType) => {
+            if (!_.has(fieldsToExtract, addressType) || !_.isEmpty(result[addressType])) {
+                return;
+            }
+            const value = addressComponent[fieldsToExtract[addressType]] ? addressComponent[fieldsToExtract[addressType]] : '';
+            result[addressType] = value;
+        });
+    });
+    return result;
```

Comment:
I'm not sure I understand what is happening here...
If we look at the example from the documentation above and the application in ApplicationSearch.js

```js
GooglePlacesUtils.getAddressComponents({long_name: "New York", short_name: "New York", types: [ "locality", "political" ]}, {
            street_number: 'long_name',
            route: 'long_name',
            locality: 'long_name',
            sublocality: 'long_name',
            postal_code: 'long_name',
            administrative_area_level_1: 'short_name',
        });
```

Then the `result` just becomes an array of empty strings that is the same size as whatever we put in as the second argument and we never update it to anything else...
![image](https://user-images.githubusercontent.com/13264774/217372710-cc057a94-5352-42ae-a04f-9224496eddb1.png)